src/dune/coq_rules.ml

```diff
@@ -192,31 +201,60 @@ let setup_ml_deps ~lib_db libs =
 let coqlib_wrapper_name (s : Dune_file.Coq.t) =
   Coq_lib_name.wrapper (snd s.name)
 
-let setup_rules ~sctx ~dir ~dir_contents (s : Dune_file.Coq.t) =
-  let scope = SC.find_scope_by_dir sctx dir in
-  let expander = SC.expander sctx ~dir in
-  if coq_debug then
-    Format.eprintf "[gen_rules] @[dir: %a@\nscope: %a@]@\n%!" Path.Build.pp dir
-      Path.Build.pp (Scope.root scope);
+let coq_modules_of_dep ~sctx lib =
+  let name = Coq_lib.name lib in
+  let dir = Coq_lib.src_root lib in
+  let dir_contents = Dir_contents.get sctx ~dir in
+  Dir_contents.coq_modules_of_library dir_contents ~name
+
+let coq_theories_of_lib_deps ~coq_lib_db libs =
+  Result.List.map ~f:(Coq_lib.DB.resolve coq_lib_db) libs |> Result.ok_exn
```

Comment:
As discussed offline, this is the only bit that needs to be improved before merging.

Reply:
Finally I've opted not to delay the resolution here (yet); this is needed for more than flags, in particular for the rules for mlpack and sources, so we gain little from delaying here (unfortunately)